File: core/registry.js
const fs = require("fs");

const REG_PATH = process.env.OMEGA_REG || "./omega_registry.json";

function load() {
    try {
        return JSON.parse(fs.readFileSync(REG_PATH));
    } catch {
        return { nodes: {}, brains: {}, lastUpdate: Date.now() };
    }
}

function save(state) {
    state.lastUpdate = Date.now();
    fs.writeFileSync(REG_PATH, JSON.stringify(state, null, 2));
}

function registerNode(id, data) {
    const state = load();
    state.nodes[id] = {
        ...data,
        lastSeen: Date.now()
    };
    save(state);
}

function registerBrain(id, data) {
    const state = load();
    state.brains[id] = {
        ...data,
        lastSeen: Date.now()
    };
    save(state);
}

module.exports = { load, save, registerNode, registerBrain };

File: core/registry.py
import json
import time

REG_PATH = "registry/node_registry.json"

def load():
    return json.load(open(REG_PATH))

def save(data):
    json.dump(data, open(REG_PATH, "w"), indent=2)

def register_node(node_id):
    data = load()

    data["nodes"][node_id] = {
        "last_seen": time.time(),
        "status": "active"
    }

    data["total_nodes"] = len(data["nodes"])
    data["active_nodes"] = len(data["nodes"])

    save(data)
